File: php/features.php
<?php
declare(strict_types=1);

// IgnavFlight SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IgnavFlightFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IgnavFlightBaseFeature();
            case "test":
                return new IgnavFlightTestFeature();
            default:
                return new IgnavFlightBaseFeature();
        }
    }
}

File: py/features.py
# IgnavFlight SDK feature factory

from feature.base_feature import IgnavFlightBaseFeature
from feature.test_feature import IgnavFlightTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IgnavFlightBaseFeature(),
        "test": lambda: IgnavFlightTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
